### PR TITLE
Add .toString() to Script class

### PR DESCRIPTION
**Pull request by erenkarakal**
### Description
Adds a .toString() method to the Script class.

Example outputs:
```
abc.sk
folder/script.sk
```

---
**Target Minecraft Versions:** any
**Requirements:** none
**Related Issues:** #5834

### CODE REVIEW

**Review by AyhamAl-Ali — APPROVED** [review]
⚡

**Comment by sovdeeth** [comment]
to sum up conversation on discord, I think the toString should hold more information about the Script object than just its file name, perhaps using the common structure of ``Script{field = value}``

**Review by Moderocky — CHANGES_REQUESTED** [review]
I don't really like this being the config file name.

File: src/main/java/ch/njol/skript/expressions/ExprPlayerProtocolVersion.java [hunk 1]
```diff
@@ -24,25 +24,49 @@
...
 	"\t\tsend \"Protocol version of %arg-1%: %protocol version of arg-1%\""})
-@Since("2.6.2")
+@Since("2.6.2, INSERT VERSION ViaVersion support")
 @RequiredPlugins("Paper 1.12.2 or newer")
 public class ExprPlayerProtocolVersion extends SimplePropertyExpression<Player, Integer> {
 
+	private static final boolean VIAVERSION_EXISTS = Skript.classExists("com.viaversion.viaversion.api.ViaAPI");
+	@Nullable
+	private static Object VIA_API;
+
```
Comment (sovdeeth) [inline src/main/java/ch/njol/skript/expressions/ExprPlayerProtocolVersion.java:43]: You've polluted this branch, this pr shouldn't have viaversion stuff.

**Review by sovdeeth — CHANGES_REQUESTED** [review]

File: src/main/java/org/skriptlang/skript/lang/script/Script.java [hunk 2]
```diff
@@ -219,4 +219,9 @@ public <T extends ScriptEvent> Set<T> getEvents(Class<T> type) {
 		);
 	}
 
+	@Override
+	public String toString() {
+		return this.getConfig().getFileName();
```
Comment (TheLimeGlass) [inline src/main/java/org/skriptlang/skript/lang/script/Script.java:224]: This should be the path string, or atleast the furthest minimum the /scripts/ directory parent as there can be different directories pertaining the same script file name.
```suggestion
		Config config = this.getConfig();
		return Optional.ofNullable(config::getPath).orElse(config::getFileName);
```

Comment (TheLimeGlass) [reply src/main/java/org/skriptlang/skript/lang/script/Script.java:224]: Would need Optional import if using suggestion.

**Comment by TheLimeGlass** [comment]
> I don't really like this being the config file name.

Do you have an alternative suggestion or you don't like the pull request. This is blocking.

File: src/main/java/ch/njol/skript/expressions/ExprPlayerProtocolVersion.java [hunk 3]
```diff
@@ -24,25 +24,49 @@
...
 			register(ExprPlayerProtocolVersion.class, Integer.class, "protocol version", "players");
 		}
+
+		if (VIAVERSION_EXISTS) {
+            try {
+                VIA_API = Class.forName("com.viaversion.viaversion.api.Via")
+					.getDeclaredMethod("getAPI")
+					.invoke(null);
+            } catch (ClassNotFoundException | NoSuchMethodException | InvocationTargetException | IllegalAccessException ignored) {}
+        }
```
Comment (Moderocky) [inline src/main/java/ch/njol/skript/expressions/ExprPlayerProtocolVersion.java:55]: Still needs the viaversion stuff removed.

File: src/main/java/org/skriptlang/skript/lang/script/Script.java (hunk 2 above)
Comment (Moderocky) [inline src/main/java/org/skriptlang/skript/lang/script/Script.java:224]: This should return something more like the IDE's default suggestion, e.g. 
```java
"Script[" +
	"config=" + config +
	", structures=" + structures +
	", suppressedWarnings=" + suppressedWarnings +
	", scriptData=" + scriptData +
	", eventHandlers=" + eventHandlers +
	']';
```

Script objects are holders for the structures and the script data, they aren't analogous to files. They also might not have a properly-linked config with a useful path if they were created by some addon at runtime. The Config toString should contain the path name, not the Script object.

**Review by Moderocky — CHANGES_REQUESTED** [review]

**Comment by sovdeeth** [comment]
Closing due to inactivity

**Pull request closed without merging**